coderd/workspaces.go

```diff
@@ -873,9 +881,16 @@ func (api *API) putWorkspaceAutostart(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	nextStartAt := sql.NullTime{}
+	if dbSched.Valid {
+		next, _ := schedule.NextAutostart(dbtime.Now(), dbSched.String, templateSchedule)
```

Comment:
What if the template schedule store does not allow this next start time?

Reply:
In the case that it is disallowed we'll end up returning a false-positive until eventually it is allowed again. This is unfortunately not ideal. 

A potential resolution for this was discussed and we'll probably have a `NextAllowedAutostart` function in a future PR that will return the first _allowed_ autostart.